### PR TITLE
Nitrox Installer

### PR DESCRIPTION
**Pull request by dm430**
A WIX 3.x based installer for Nitrox.

Dependent on:
- WIX Toolset
- dnlib

### CODE REVIEW

File: Nitrox.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.25420.1
+# Visual Studio 15
```
Comment (Sunrunner37) [inline Nitrox.sln]: Is it possible to leave this file untouched?

File: NitroxInstallerActions/packages.config [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="dnlib" version="1.6.1" targetFramework="net461" />
```
Comment (Sunrunner37) [inline NitroxInstallerActions/packages.config]: Should we add a git submodule for dnlib? (https://github.com/0xd4d/dnlib)

**Comment by AlexejheroYTB** [comment]
What would this installer compile into? (.exe?). And would it need to have the dll files in the same folder as it? (So an archive is required?)

File: Nitrox.sln (hunk 1 above)
Comment (dm430) [reply Nitrox.sln]: Yes, I just figured we would want the projects to be part of the solution for easy management. If not I can revert this file.

File: NitroxInstallerActions/packages.config (hunk 2 above)
Comment (dm430) [reply NitroxInstallerActions/packages.config]: That is up to you. If you want I will add it as a sub module.

**Comment by dm430** [comment]
@AlexjheroYTB it is a self contained MSI built with the windows installer xml (WIX) toolset. This allows for full control and customization of the installer.

**Comment by AlexejheroYTB** [comment]
Does it need to have the other dll files in the same folder/archive to run, or are they archived/compiled in it?

**Pull request merged**